### PR TITLE
feat: cluster errors

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

Building on https://github.com/PostHog/posthog/pull/20752

We seem to be able to cluster replay errors effectively. Adds a method to do just that for

## Changes

General approach:
- Fetch all embedded errors for the last week
- Run clustering algorithm
- Add cluster labels to original data
- Construct summary for display on the frontend

Also adds timings to get a sense of clustering time (although it will most likely be on a background task)

## How did you test this code?

Ran things locally with some test data

### CODE REVIEW

File: ee/session_recordings/ai/error_clustering.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+from django.conf import settings
+from posthog.clickhouse.client import sync_execute
+from posthog.models.team import Team
+from sklearn.cluster import DBSCAN
+import pandas as pd
+
+CLUSTER_REPLAY_ERRORS_TIMING = Histogram(
+    "posthog_session_recordings_cluster_replay_errors",
+    "Time spent clustering the embeddings of replay errors",
+)
```
Comment (pauldambra) [inline ee/session_recordings/ai/error_clustering.py:11]: should we have specific buckets here? is this likely slower than an API call?

**Review by pauldambra — APPROVED** [review]
🍧

File: ee/session_recordings/ai/error_clustering.py (hunk 1 above)
Comment (daibhin) [reply ee/session_recordings/ai/error_clustering.py:11]: Yep, good catch. Much more so - I'd expect the clustering to be measured in seconds
